File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/Basic/RemotingHello/Client/consoleclientall/HelloAsync.cs
/*=====================================================================

  File:      HelloAsync.cs

---------------------------------------------------------------------
This file is part of the Microsoft .NET Framework SDK Code Samples.

  Copyright (C) Microsoft Corporation.  All rights reserved.

This source code is intended only as a supplement to Microsoft
Development Tools and/or on-line documentation.  See these other
materials for detailed information regarding Microsoft code samples.
 
THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
PARTICULAR PURPOSE.
=====================================================================*/

using System;
using System.Runtime.Remoting;
using System.Runtime.Remoting.Messaging;
using System.Threading;

using Hello;

public class HelloAsync
{
    public static void Main(String[] args)
    {
        Console.WriteLine(".NET Remoting Sample - HelloAsync");
        Console.WriteLine("---------------------------------\n");

        HelloAsync helloAsync = new HelloAsync();
        helloAsync.Run();
    }

    public void Run()
    {
        String name = "Bill";    
        String configFilename = "HelloAsync.exe.config";

        Console.WriteLine("Configuring Remoting from {0}", configFilename);
        RemotingConfiguration.Configure(configFilename);

        Console.WriteLine("Obtaining Proxy for HelloService, using new");
        HelloService helloService = new HelloService();

        Console.WriteLine("Creating ReturnGreetingDelegate to HelloService.ReturnGreeting");
        ReturnGreetingDelegate returnGreetingDelegate = new ReturnGreetingDelegate(helloService.ReturnGreeting);

        Console.WriteLine("Calling returnGreetingDelegate.BeginInvokeInvoke");
        IAsyncResult ar = returnGreetingDelegate.BeginInvoke(name, null, null);

        Console.WriteLine("Calling returnGreetingDelegate.EndInvoke");
        String greeting = returnGreetingDelegate.EndInvoke(ar);

        Console.WriteLine("returnGreetingDelegate.EndInvoke returned: {0}", greeting);
    }

    public delegate String ReturnGreetingDelegate(String name);

}

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/VB/Basic/RemotingHello/Client/ConsoleClientAll/helloasync.vb

Imports System
Imports System.Runtime.Remoting
Imports System.Runtime.Remoting.Messaging
Imports System.Threading
Imports Microsoft.VisualBasic

Imports Hello



Public Class HelloAsync
    
    Public Shared Sub Main()
        Console.WriteLine(".NET Remoting Sample - HelloAsync")
        Console.WriteLine("---------------------------------" + ControlChars.Cr)
        
        Dim helloAsync As New HelloAsync()
        helloAsync.Run()
    End Sub 'Main
    
    
    Public Sub Run()
        Dim name As String = "Bill"
        Dim configFilename As String = "HelloAsync.exe.config"
        
        Console.WriteLine("Configuring Remoting from {0}", configFilename)
        RemotingConfiguration.Configure(configFilename)
        
        Console.WriteLine("Obtaining Proxy for HelloService, using new")
        Dim helloService As New HelloService()
        
        Console.WriteLine("Creating ReturnGreetingDelegate to HelloService.ReturnGreeting")
        Dim returnGreetingDelegate As New ReturnGreetingDelegate(AddressOf helloService.ReturnGreeting)
        
        Console.WriteLine("Calling returnGreetingDelegate.BeginInvokeInvoke")
        Dim ar As IAsyncResult = returnGreetingDelegate.BeginInvoke(name, Nothing, Nothing)
        
        Console.WriteLine("Calling returnGreetingDelegate.EndInvoke")
        Dim greeting As String = returnGreetingDelegate.EndInvoke(ar)
        
        Console.WriteLine("returnGreetingDelegate.EndInvoke returned: {0}", greeting)
    End Sub 'Run
    
    
    Delegate Function ReturnGreetingDelegate(name As String) As String
End Class 'HelloAsync 
